.github/workflows/codeql.yaml

```
@@ -2,22 +2,46 @@ name: Code Scanning
 
 on:
   push:
+    branches: [ "main" ]
   pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
```

IMHO we should narrow down the types of events (see the full list here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request=)

```suggestion
    branches:
      -  main
    types: [ opened, synchronize, reopened ]
```

---

Yep, but what I checked by default it uses only `opened, synchronize, reopened`. I also tested that by adding the label. Also your comments didn't trigger the pipeline. IMO it's better to stay with the defaults in this case.